File: dataset_reduzido/avatar/Python/Code/atcoder_ABC145_B.py
n = int ( input ( ) )
s = input ( )
if n % 2 != 0 : print ( 'No' )
elif s [ : int ( n / 2 ) ] == s [ int ( n / 2 ) : ] : print ( 'Yes' )
else : print ( 'No' )

File: output/StarCoder/avatar/Python/Java/atcoder_ABC145_B.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        scanner.nextLine(); // Consume newline left-over
        String s = scanner.nextLine();
        
        if (n % 2 != 0) {
            System.out.println("No");
        } else if (s.substring(0, n / 2).equals(s.substring(n / 2))) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
        
        scanner.close();
    }
}